File: app/src/main/java/com/example/chris/atp_music_player/ui/LibraryFragment.java
package com.example.chris.atp_music_player.ui;

import android.os.Bundle;
import android.support.design.widget.TabLayout;
import android.support.v4.app.Fragment;
import android.support.v4.view.ViewPager;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.example.chris.atp_music_player.R;
import com.example.chris.atp_music_player.adapters.LibraryPagerAdapter;

import butterknife.ButterKnife;
import butterknife.InjectView;

public class LibraryFragment extends Fragment {
    private final String PAGE_NUM = "page_num";

    @InjectView(R.id.library_viewpager)
    ViewPager mViewPager;
    @InjectView(R.id.library_tabs)
    TabLayout mPagerTabs;

    public static LibraryFragment newInstance() {
        return new LibraryFragment();
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_library, container, false);
        ButterKnife.inject(this, view);

        getActivity().setTitle(R.string.menu_lib);

        initPager(savedInstanceState);

        return view;
    }

    @Override
    public void onSaveInstanceState(Bundle outState) {
        super.onSaveInstanceState(outState);
        outState.putInt(PAGE_NUM, mViewPager.getCurrentItem());
    }

    public void initPager(Bundle bundle) {
        LibraryPagerAdapter adapter = new LibraryPagerAdapter(getActivity(),
                getChildFragmentManager());

        mViewPager.setAdapter(adapter);

        if (null != bundle) {
            mViewPager.setCurrentItem(bundle.getInt(PAGE_NUM));
        }

        mPagerTabs.setupWithViewPager(mViewPager);
    }
}

File: app/src/main/java/com/example/chris/atp_music_player/ui/LibraryFragment.kt
package com.example.chris.atp_music_player.ui

import android.os.Bundle
import android.support.v4.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup

import com.example.chris.atp_music_player.R
import com.example.chris.atp_music_player.adapters.LibraryPagerAdapter

import kotlinx.android.synthetic.main.fragment_library.*

class LibraryFragment : Fragment() {
    private val PAGE_NUM = "page_num"

    override fun onCreateView(inflater: LayoutInflater?, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        val view = inflater!!.inflate(R.layout.fragment_library, container, false)
        activity.setTitle(R.string.menu_lib)
        return view
    }

    override fun onViewCreated(view: View?, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        library_viewpager.adapter = LibraryPagerAdapter(resources.getStringArray(R.array.library_pager_items),
                childFragmentManager)
        library_viewpager.currentItem = if (savedInstanceState != null) savedInstanceState.getInt(PAGE_NUM) else 0
        library_tabs.setupWithViewPager(library_viewpager)
    }

    override fun onSaveInstanceState(outState: Bundle?) {
        super.onSaveInstanceState(outState)
        outState!!.putInt(PAGE_NUM, library_viewpager.currentItem)
    }
}
